libs/core/src/components/pds-select/pds-select.tsx

```diff
@@ -0,0 +1,207 @@
+   */
+  @Prop() name!: string;
+
+  /**
+   * Indicates whether or not the select field is required.
+   */
+  @Prop() required: boolean;
+
+  /**
+   * The value(s) of the selected option(s).
+   *
+   */
+  @Prop({ mutable: true }) value?: string | string[];
+
+  /**
+   * Emitted when a keyboard input occurred.
+   */
+  @Event() pdsSelectChange: EventEmitter<InputEvent>;
+
+  @Watch('value')
+  /**
+   * Handles the change in the value of the select component.
+   * This method is called whenever the value of the select component changes.
+   * It updates the selected option accordingly.
+   */
+  valueChanged() {
+    this.updateSelectedOption();
+  }
+
+  /**
+   * Lifecycle method that is called just before the component is loaded.
+   * This method is used to perform any necessary updates to the selected option
+   * before the component is rendered.
+   *
+   * @returns {void}
+   */
+  componentWillLoad() {
+    this.updateSelectedOption();
+  }
+
+  /**
+   * Updates the selected option in the select element based on the current value.
+   *
+   * This method iterates through all the options of the select element and sets the
+   * 'selected' attribute on the option that matches the current value. If an option
+   * does not match the current value, the 'selected' attribute is removed.
+   *
+   * @private
+   * @returns {void}
+   */
+  private updateSelectedOption() {
+    if (this.selectEl) {
+      const options = this.selectEl.options;
+
+      // Update the selected attribute for all options
+      Array.from(options).map((option: HTMLOptionElement) => {
+        if (Array.isArray(this.value)) {
+          option.selected = this.value.includes(option.value);
+        } else {
+          option.selected = this.value === option.value;
+        }
+      });
+    };
+  }
+
+  /**
+   * Emits an event on input change
+  */
+  private onSelectUpdate = (e: Event) => {
+    const target = e.target as HTMLSelectElement
+
+    const values = Array.from(target.options)
+        .filter((option) => ( option.selected))
+        .map((option) => ( option.value));
+
+    if (values.length === 1 && !this.multiple) {
+        this.value = values[0];
+    } else {
+        this.value = values;
+    }
+
+    this.pdsSelectChange.emit(e as InputEvent);
+  };
+
+  /**
+   * Handles the change event for the slot element.
+   * This method is triggered when the slot content changes.
+   * It updates the inner HTML of the select element by cloning and appending
+   * the assigned <option> elements from the slot.
+   */
+  private handleSlotChange = () => {
+    const slot = this.slotContainer.querySelector('slot') as HTMLSlotElement;
+
+    this.selectEl.innerHTML = '';
+    const assignedElements = slot.assignedElements({ flatten: true }) as (HTMLOptionElement | HTMLOptGroupElement)[];
+
+    assignedElements.forEach((item) => {
+      if (item.tagName === 'OPTION' || item.tagName === 'OPTGROUP') {
```

Comment:
it looks like there was a change that was lost.
```suggestion
      if ( ['OPTION', 'OPTGROUP'].includes(item.tagName)) {
```

Reply:
Done. Thaks for the suggestion. Not sure where it went.